### PR TITLE
[WIP] github: Execute tasks in parallel to fully utilize runner

### PR DESCRIPTION
**Pull request by davidkupka**
DON'T EXECUTE THIS. It's free-hand coding and it's guaranteed to misbehave.
This is my idea for executing tasks in parallel which would let us utilize beefy runners fully.

### CODE REVIEW

**Comment by tomaskrizek** [comment]
The approach looks good. Are you aware of anything that's still missing?

**Comment by davidkupka** [comment]
1. All jobs in .freeipa-pr-ci.yaml should be extended with
```
resources:
    cpu: N  # number of CPU cores needed by job
    memory: M  # amount of RAM needed by job (in bytes)
```
otherwise it's assumed that the job needs whole runner and only one task at the time is executed.

2. Handle termination better. Currently, first SIGINT sets Executive.done to True meaning that it won't start any new jobs and will wait for running ones to finish. Second SIGINT calls Executive.abort which sends SIGTERM to all child processes and wait for them to finish. Third SIGINT calls sys.exit. It sort of makes sense but only if subprocesses handle SIGTERM as a request to abort and clean up immediately.

3. TASK_MIN_CPU and TASK_MIN_MEM should be moved to configuration file. Those variables define the minimal amount of resources left on runner when it makes sense to look for jobs to execute.

File: github/prci.py [hunk 1]
```diff
@@ -211,32 +179,109 @@ def sentry_report_exception(context=None):
...
+        self.done = True
+        self.reboot = False
+
+    def abort(self):
+        if self.abort:
+            sys.exit('Forced quit. There may be stale files and processes '
+                     'left behind.')
+        logging.debug('Abort. Waiting for clean up.')
+        self.done = True
+        self.abort = True
```
Comment (tomaskrizek) [inline github/prci.py]: This should also call terminate() for all processes.

File: github/prci_github/internals.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 import yaml
 
 import cachecontrol
+import psutil
```
Comment (tomaskrizek) [inline github/prci_github/internals.py]: Add psutil to ansible playbook dependencies.

**Comment by felipevolpone** [comment]
@davidkupka thank you very much for the patch :)) I've merged into the code and changed a little if you don't mind.  PR #114 address it.

**Comment by tomaskrizek** [comment]
We've decided to abandon this approach, as it proved to be very error prone and we don't want to break the existing infrastructure. See https://github.com/freeipa/freeipa-pr-ci/issues/54#issuecomment-344917531 for details.

**Pull request closed without merging**